### DISCUSSION TITLE
Renovate automerges before Github Actions CI starts running

### DISCUSSION THREAD

**Discussion by mikael-mattsson** [renovatebot/renovate#13982] [Request Help]
We have been noticing that sometimes Github hosted Renovate automerges updates that have failed in our Github Actions test and/or lint jobs.

This is our Renovate config:
```
{
  "timezone": "Europe/Stockholm",
  "schedule": ["after 8am and before 5pm every weekday"],
  "extends": [
    "config:js-app",
    ":disableRateLimiting",
    ":rebaseStalePrs",
    "npm:unpublishSafe",
    ":automergeRequireAllStatusChecks",
    ":automergeBranch",
    ":automergeMinor",
    ":automergePatch",
    ":maintainLockFilesWeekly"
  ]
}
```

After doing some digging, it seems like `npm:unpublishSafe` makes Renovate immediately merge the update if the `renovate/stability-days` check is succeeded instantly upon branch creation and our Github Actions CI jobs haven't started yet.

As can be seen around line 7677 and onward in the attached [Renovate log file](https://github.com/renovatebot/renovate/files/7995470/renovate.log), `resolveBranchStatus` runs three times, and already at the first run `branch status check result` shows that the stability check has succeeded. Then `No check runs found` is printed, indicating no Github Actions CI has started yet. After three `resolveBranchStatus` checks the branch is merged.

However, if we call the list check-runs endpoint ourselves today, we can see that our CI has been run and that the lint job has failed. We can also see that the job started 18 seconds after the stability days check was last updated.

```
{
  "id": ***redacted***,
  "name": "Lint (14, ubuntu-latest)",
  "node_id": "***redacted***",
  "head_sha": "***redacted***",
  "external_id": "***redacted***",
  "url": "***redacted***",
  "html_url": "***redacted***",
  "details_url": "***redacted***",
  "status": "completed",
  "conclusion": "failure",
  "started_at": "2022-02-02T13:36:34Z",
  "completed_at": "2022-02-02T13:38:38Z",
.
.
.
```

What's going on here? Have we discovered a bug or is there a problem with our Renovate config?

**Comment by rarkins** [comment]
I think the best thing you can do is add GitHub's branch protection with required status checks. That way this scenario would be impossible to begin with because GitHub would block any attempts to merge. Is that an option for you?

**Reply by mikael-mattsson** [reply]
Okay, there's no way to configure Renovate to wait for a number of minutes before a merge is allowed? Like, let's say the branch must have been alive for at least 10 minutes?

**Reply by viceice** [reply]
Nope

**Reply by mikael-mattsson** [reply]
Alright, I'll have a look at branch protection then. Thanks for taking your time to answer!
